src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,99 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+import PropTypes from 'prop-types';
+
+import { getData, post, remove } from '../../api/api';
+
 import { NewCommentForm } from '../NewCommentForm';
+
 import './PostDetails.scss';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+export const PostDetails = ({ selectedPostId }) => {
+  const [postDetails, setPostDetails] = useState(0);
+  const [isVisibleComments, setVisibleComments] = useState(true);
+  const [comments, setComments] = useState([]);
+
+  const addComment = (name, email, body) => {
+    post('/comments', {
+      name, email, body, postId: selectedPostId,
+    }).then((newComment) => {
+      setComments(listOfComments => [newComment, ...listOfComments]);
+    });
+  };
+
+  const removeComment = async(commentId) => {
+    await remove(`/comments/${commentId}`);
+    const commentsSever = await getData(`/comments?postId=${selectedPostId}`);
+
+    setComments(commentsSever);
+  };
+
+  const getPostDetails = async(selectedPost) => {
+    const postFromServer = await getData(`/posts/${selectedPost}`);
```

Comment:
These functions (addComment, removeComment and getPostDetails) should be in the `comments.js` file (as task description requires)